engine/resource/src/resource_mounts.cpp

```diff
@@ -0,0 +1,539 @@
+    if (visited)
+    {
+        *visited = exists;
+        return true;
+    }
+
+    if (t.Full())
+    {
+        uint32_t capacity = t.Capacity();
+        capacity += 32;
+        uint32_t table_size = (capacity*2)/3;
+        t.SetCapacity(table_size, capacity);
+    }
+    t.Put(url_hash, exists);
+    return exists; // it didn't exist
+}
+
+static dmResource::Result GetDependenciesInternal(DependencyIterContext* ctx, const dmhash_t url_hash)
+{
+    ResourceMountsContext* mounts_ctx = (ResourceMountsContext*)ctx->m_Mounts;
+    DM_MUTEX_SCOPED_LOCK(mounts_ctx->m_Mutex);
+
+    uint32_t num_mounts = mounts_ctx->m_Mounts.Size();
+    bool only_missing = ctx->m_OnlyMissing;
+
+    dmArray<dmhash_t> dependencies; // allocate it outside of the loop
+
+    for (uint32_t i = 0; i < num_mounts; ++i)
+    {
+        ArchiveMount& mount = mounts_ctx->m_Mounts[i];
+
+        dmResource::Manifest* manifest;
+        dmResourceProvider::Result presult = dmResourceProvider::GetManifest(mount.m_Archive, &manifest);
+        if (presult != dmResourceProvider::RESULT_OK)
+            continue;
+
+        dependencies.SetSize(0);
+        dmResource::Result result = dmResource::GetDependencies(manifest, url_hash, dependencies);
+        if (dmResource::RESULT_RESOURCE_NOT_FOUND == result)
+            continue;
+
+        uint32_t hash_len = dmResource::GetEntryHashLength(manifest);
+
+        for (uint32_t d = 0; d < dependencies.Size(); ++d)
+        {
+            dmhash_t dep_hash = dependencies[i];
+            bool* visited = ctx->m_Visited.Get(dep_hash);
+            if (visited)
+                continue;
+
+            bool exists = true;
+            if (only_missing)
+            {
+                exists = ResourceExists(mounts_ctx, dep_hash) == dmResource::RESULT_OK;
+            }
+
+            UpdateVisited(ctx->m_Visited, dep_hash, exists);
+
+            // If we only want the missing ones
+            if (only_missing && exists)
+                continue;
+
+
+            SGetDependenciesResult result = {0};
+            result.m_UrlHash = dep_hash;
+            result.m_Missing = !exists;
+            dmLiveUpdateDDF::ResourceEntry* entry = dmResource::FindEntry(manifest, dep_hash);
+            if (!entry)
+            {
+                // NOTE: It is important that we don't remove "liveupdate" resource dependencies from the manifest
+                // Otherwise we cannot get the actual path to report back to the user for the download
+                dmLogError("Couldn't get entry for dependency %llx in manifest from ´%llx`", dep_hash, url_hash);
+                ctx->m_UserCallback(ctx->m_UserCallbackCtx, &result);
+                continue;
+            }
+
+            result.m_HashDigest        = entry->m_Hash.m_Data.m_Data;
+            result.m_HashDigestLength  = hash_len;
+
+            ctx->m_UserCallback(ctx->m_UserCallbackCtx, &result);
+
+            if (ctx->m_Resursive && entry->m_Dependants.m_Count > 0)
```

Comment:
Our code now supports recursive lookup.
We need to add a flag to the Lua api though.